test/test.js

```diff
@@ -10,6 +10,12 @@ function sortLeftToRight(a, b) {
 	return 0;
 }
 
+function assertTree(tree, nbElements) {
+	assert.strictEqual(tree._getCount(tree.root), nbElements);
+	assert.strictEqual(tree._isSorted(tree.root), true);
```

Comment:
I know that isSorted and isBalanced could be part of the module only. But I wanted to be able to access them in the code where the trees will be implemented (for debugging purpose).
Maybe I shouldn't, I will keep thinking about this.

About your tests, I had to replace them, because changing the implementation made some of your tests fail, although the result was valid with respect to the specifications of an AVL tree (balanced and sorted). I do not think I will go as far as testing the tests. The problems with the tests you wrote is that they tested implementation and not features. Although we want to be flexible in terms of how we implement our component.